### PR TITLE
DROOLS-5950 Support returning PackageDescr from assemblers

### PR DESCRIPTION
**Pull request by evacchi**
**JIRA**: https://issues.redhat.com/browse/DROOLS-5950

**referenced Pull Requests**: 

- https://github.com/kiegroup/drools/pull/3335
- https://github.com/kiegroup/droolsjbpm-knowledge/pull/497

### CODE REVIEW

File: kie-internal/src/main/java/org/kie/internal/services/KieAssemblersImpl.java [hunk 1]
```diff
@@ -39,6 +39,19 @@ public KieAssemblersImpl() {
         return assemblers;
     }
 
+    @Override
+    public void addResourceAsPackageDescr(Object knowledgeBuilder, Resource resource, ResourceType type, ResourceConfiguration configuration) throws Exception {
```
Comment (gitgabrio) [inline kie-internal/src/main/java/org/kie/internal/services/KieAssemblersImpl.java:43]: @evacchi 
Sorry, I can't get that. There is a method 
`addResourceAsPackageDescr` whose name clearly suggest the added resource is a `PackageDescr` (and only that).
But this method invoke `addResourceBeforeRules` whose name (and semantic) is different, i.e. define to add a resource (regardless of generated type) before rules are compiled. 
Just to be overtly pedantic, the former method has a "what" semantic, while the latter has a "when" semantic.
I think this is pretty misleading. Beside that, I think that the "what"  should be an implementation detail, and that all those methods should only refer to the "when".
Long story short: could you please rename  `addResourceAsPackageDescr` to `addResourceBeforeRules` ? This would make it consistent and clear everywhere it is read/used

**Review by gitgabrio — CHANGES_REQUESTED** [review]

File: kie-internal/src/main/java/org/kie/internal/services/KieAssemblersImpl.java (hunk 1 above)
Comment (evacchi) [reply kie-internal/src/main/java/org/kie/internal/services/KieAssemblersImpl.java:43]: yes. I actually renamed the other one too and only forgot this one :P

Comment (evacchi) [reply kie-internal/src/main/java/org/kie/internal/services/KieAssemblersImpl.java]: (see previous commit :P)

Comment (gitgabrio) [reply kie-internal/src/main/java/org/kie/internal/services/KieAssemblersImpl.java]: :+1:

**Review by gitgabrio — APPROVED** [review]

**Comment by evacchi** [comment]
jenkins run cdb

**Review by hellowdan — APPROVED** [review]

**Review by mariofusco — APPROVED** [review]

**Comment by evacchi** [comment]
jenkins run cdb
